pkg/controller/storagecluster/reconcile.go

```diff
@@ -530,6 +530,11 @@ func determinePlacementRack(nodes *corev1.NodeList, node corev1.Node, minRacks i
 	}
 
 	sort.Strings(rackList)
+	if len(rackList) == 0 {
```

Comment:
instead of `len(rackList) == 0`, I recommend ` len(targetAZ) > 0`:

```
if len(targetAZ) > 0 {
    # fill rackList with nodes from targetAZ like before
} else {
    # just fill rackList with all the nodes like you're doing in the PR
}
```

Reply:
done